sentry-kotlin-multiplatform/src/androidMain/kotlin/io/sentry/kotlin/multiplatform/SentryScope.kt

```diff
@@ -0,0 +1,114 @@
+package io.sentry.kotlin.multiplatform
+
+import io.sentry.ScopeCallback
+import io.sentry.kotlin.multiplatform.extensions.*
+
+actual class SentryScope : ISentryScope {
+
+    // We are directly modifying the Android SDK scope
+    private var scope: AndroidSentryScope? = null
+
+    actual override var level: SentryLevel?
+        set(value) {
+            scope?.level = value?.toAndroidSentryLevel()
+        }
+        get() {
+            return scope?.level?.toKMPSentryLevel()
+        }
+
+    actual override var user: SentryUser?
+        set(value) {
+            value?.onFieldChanged = {
```

Comment:
Why do we need `onFieldChanged`? can we just call it directly?

Reply:
if for example the user does something like this:

```Kotlin
configureScope { scope ->
  scope.user = User()
  scope.user.username = "username"
}

Then the set(value) is not triggered during 'user.username = ' so I put in a callback to `SentryUser` fields so we know when to update the scope's user. Maybe there's a much better way of doing this? maybe pass the scope to User and modify it directly there?

Reply:
the goal is to do:
`val myUser = scope.user`
`scope.user = myUser`